### ISSUE TITLE
Unit settings change automatically even when driving region is not in automatic mode

### ISSUE THREAD

**Issue by kodiakcreativity** [osmandapp/OsmAnd#15188]
### Description
Driving region has a setting for automatic, but I have set the region to Canada. I want the units of OSMAND to always be metric regardless of where I am, even if I am in the US. While driving in the US the units keep reverting back to miles/feet and I have to constantly change it back to km/m. Seems that crossing state lines may have something to do with it, this prompts the app to identify what driving region I am in and it changes the region automatically and the units along with it even though I specifically turned automatic region off.

### Your Environment
OsmAnd Version: 4.1.11
Android/iOS version:12
Device model:S22 ultra

**Maps used (online or offline):**   
If you have an issue related to offline maps, tell us the exact name of the map file where the issue occurs and its edition date.

**Comment by tradavyvy** [comment]
👍
I have the same problem.
The whole yard, inch, mile, feet, acre, city block, square cubit measurement system is crazy and I keep reverting osmand to decimal whenever I drive through the us.
(I admit that I secretly stil use the 60 seconds, 60 minute, 24 hour, 7 day, 12 month system that in itself is only marginaly more logical 🙃)

**Comment by Zirochkabila** [comment]
Testing was performed on OsmAnd application version 4.1.11, released: 2022-01-19, and OsmAnd 4.2.7, released: 2022-07-13 - it wasn't possible to reproduce. The Canadian region and units are retained after the Canada-U.S. boundary is crossed.

Try resetting the profile to default and try again: Menu > Configure profile > Reset to default
Or reinstall the application but first make a backup: Menu > Settings > Backup & Restore

If you still find it, please send us the log file (please save the log file as soon as you pinpointed it): Menu > Help > Send logcat log

>

**Comment by kodiakcreativity** [comment]
Thank you so much for looking into this!
My observation seems to show it may be triggered by crossing state lines within the US, but that's just a guess. The other factor though is this is not an immediate thing. I don't think I've ever seen it change the units instantly when crossing state lines, actually, I don't think I've seen this change happen at all while using navigation.
When I set the units to metric they will stay there for a time, up to several days. Then one day out of nowhere I'll launch OSMAND and start navigating and suddenly realize its reading me distances in miles instead of kilometers. (again seems I usually crossed a state line recently when I see this happen, but I cross state lines quite frequently in my travels, so it could just be coincidence) If can pinpoint a time when I know for a fact it was in metric the last time I used it and suddenly find it to be in imperial/SAE I will capture that log as you suggest and post it here.
thank you again!
